File: swea_1208.java
/**
 * 
 */
package algorithms;

import java.util.Arrays;
import java.util.Scanner;

/**
 * @author leejuhyun
 * SWEA no.1208: Flatten
 * 
 * memory: 33,788kb
 * execution time: 247ms
 * 
 */
public class swea_1208 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		for(int idx=1; idx < 11; idx++) {
			int n = sc.nextInt(); // number of dumping
			int cnt = 0;
			int[] heights = new int[100];
			for (int i=0; i<100; i++) {
				heights[i] = sc.nextInt();
			}
			
			int max_height = Arrays.stream(heights).max().getAsInt();
			int min_height = Arrays.stream(heights).min().getAsInt();
			
			while (cnt < n) {
				if (max_height - min_height <= 1) {
					break;
				}
				int max_index = findIndex(heights, max_height);
				heights[max_index] -= 1;
				int min_index = findIndex(heights, min_height);
				heights[min_index] += 1;
				cnt += 1;
				
				max_height = Arrays.stream(heights).max().getAsInt();
				min_height = Arrays.stream(heights).min().getAsInt();		
			}
			int diff = max_height - min_height;
			System.out.printf("#%d %d", idx, diff);
			System.out.println();
		}
		
	

	}public static int findIndex(int[] arr, int target) {
		for (int i=0; i < arr.length; i++) {
			if (arr[i] == target) {
				return i;
			}
		}
		return -1;

	}

}

File: SWEA/swea_1208.py
'''
SWEA 1208.Flatten
'''
for idx in range(1, 11):
    n = int(input()) # 덤프 횟수
    cnt = 0
    heights = list(map(int, input().split())) # 100개의 높이가 있는 리스트
    max_height = max(heights)
    min_height = min(heights)

    while (cnt < n):
        if max_height - min_height <= 1:
            break
        max_index = heights.index(max_height)
        heights[max_index] -= 1
        min_index = heights.index(min_height)
        heights[min_index] += 1
        cnt += 1
        max_height = max(heights)
        min_height = min(heights)

    diff = max_height - min_height
    print(f'#{idx} {diff}')
